pkgs/os-specific/linux/hid-fanatecff/default.nix

```
@@ -0,0 +1,38 @@
+{ lib, stdenv, kernel, fetchFromGitHub, }:
+
+stdenv.mkDerivation rec {
+  pname = "hid-fanatecff";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "gotzl";
+    repo = "hid-fanatecff";
+    rev = "0.1";
+    hash = "sha256-1Nm/34Er/qfel9LJp++IWd7cTh2Wi93Kgd28YLMVvWo=";
+  };
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  makeFlags = kernel.makeFlags ++ [
+    "KVERSION=${kernel.modDirVersion}"
+    "KERNEL_SRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+  ];
+
+  installPhase = ''
+    mkdir -p $out/etc/udev/rules.d
+    cp ${src}/fanatec.rules $out/etc/udev/rules.d/99-fanatec.rules
+
+    mkdir -p $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/hid
+    cp hid-fanatec.ko $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/hid
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/etc/udev/rules.d
    cp ${src}/fanatec.rules $out/etc/udev/rules.d/99-fanatec.rules

    mkdir -p $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/hid
    cp hid-fanatec.ko $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/hid
    
    runHook postInstall
  '';
```

Don't forget to also add preInstall / postInstall hooks in the installPhase :)

See https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#phases-sec-stdenv-phases

> When overriding a phase, for example installPhase, it is important to start with runHook preInstall and end it with runHook postInstall, otherwise preInstall and postInstall will not be run. Even if you don't use them directly, it is good practice to do so anyways for downstream users who would want to add a postInstall by overriding your derivation.